### PR TITLE
Fix undefined references in mingw gthr

### PR DESCRIPTION
**Pull request by eukarpov**
```
/home/user/cross/bin/aarch64-w64-mingw32-gcc -O2 -DMS_ABI -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=20 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.25.dev\" -march=armv8-a -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  libopenblas.def dllinit.obj \
-shared -o ../libopenblas.dll -Wl,--out-implib,../libopenblas.dll.a \
-Wl,--whole-archive ../libopenblas_armv8p-r0.3.25.dev.a -Wl,--no-whole-archive  -defaultlib:advapi32 -lgfortran -defaultlib:advapi32 -lgfortran
/home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/../../../../aarch64-w64-mingw32/bin/ld: /home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/libgfortran.a(error.o):(.rdata$.refptr.__gthr_win32_self[.refptr.__gthr_win32_self]+0x0): undefined reference to `__gthr_win32_self'
/home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/../../../../aarch64-w64-mingw32/bin/ld: /home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/libgfortran.a(async.o):(.rdata$.refptr.__gthr_win32_join[.refptr.__gthr_win32_join]+0x0): undefined reference to `__gthr_win32_join'
/home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/../../../../aarch64-w64-mingw32/bin/ld: /home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/libgfortran.a(async.o):(.rdata$.refptr.__gthr_win32_create[.refptr.__gthr_win32_create]+0x0): undefined reference to `__gthr_win32_create'
/home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/../../../../aarch64-w64-mingw32/bin/ld: /home/user/cross/lib/gcc/aarch64-w64-mingw32/14.0.0/libgfortran.a(read.o):(.rdata$.refptr.__strtold[.refptr.__strtold]+0x0): undefined reference to `__strtold'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:124: ../libopenblas.dll] Error 1
make[1]: Leaving directory '/home/user/workspace/OpenBLAS/exports'
make: *** [Makefile:150: shared] Error 2
```

### CODE REVIEW

File: libgfortran/libgfortran.h [hunk 1]
```diff
@@ -67,8 +67,14 @@ extern float __strtof (const char *, char **);
 #define gfc_strtof __strtof
 extern double __strtod (const char *, char **);
 #define gfc_strtod __strtod
+
 extern long double __strtold (const char *, char **);
+#if defined(__arm__) || defined(__aarch64__) || defined(_ARM_) || defined(_ARM64_)
```
Comment (Blackhex) [inline libgfortran/libgfortran.h]: This fix seems to be unrelated to the other one (which is equivalent to [https://github.com/ZacWalk/gcc-woarm64/pull/5/files](https://github.com/ZacWalk/gcc-woarm64/pull/5/files)), i.e., it would deserve a separate PR.

Comment (Blackhex) [inline libgfortran/libgfortran.h]: It shoud be enough to test only one macro, I guess the `__aarch64__` is the preferred one, but please do your own rearch whether this is true.

File: libgfortran/libgfortran.h [hunk 2]
```diff
@@ -67,8 +67,14 @@ extern float __strtof (const char *, char **);
 #define gfc_strtof __strtof
 extern double __strtod (const char *, char **);
 #define gfc_strtod __strtod
+
```
Comment (Blackhex) [inline libgfortran/libgfortran.h]: Did you try to change line 65 from:

```
#ifdef __MINGW32__
```

to 

```
#if defined(__MINGW32__) && ! defined(__aarch64__)
```

I guess, it might be a simpler fix of the issue, in case it works, ofc (haven't tried).

Comment (eukarpov) [reply libgfortran/libgfortran.h]: Moved to a separate PR
https://github.com/ZacWalk/gcc-woarm64/pull/10

File: libgfortran/libgfortran.h (hunk 1 above)
Comment (eukarpov) [reply libgfortran/libgfortran.h]: Moved to a separate PR
https://github.com/ZacWalk/gcc-woarm64/pull/10

Comment (eukarpov) [reply libgfortran/libgfortran.h]: Moved to a separate PR
https://github.com/ZacWalk/gcc-woarm64/pull/10

**Review by Blackhex — APPROVED** [review]
As I have experimented whit those changes as well, I can approve.

**Comment by eukarpov** [comment]
PR is validated in a CI here
https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/runs/7004817860
